sql/core/src/test/scala/org/apache/spark/sql/DateFunctionsSuite.scala

```diff
@@ -69,11 +70,26 @@ class DateFunctionsSuite extends QueryTest with SharedSQLContext {
     checkAnswer(
       df.select("t").filter($"t" <= "2014-06-01"),
       Row(Timestamp.valueOf("2014-01-01 00:00:00")) :: Nil)
+  }
 
+  test("timestamp comparison with date strings with session local timezone") {
```

Comment:
it seems we are using "timestamp string" in this test, shall we update the test name?